src/timer.cc

```diff
@@ -76,11 +79,32 @@ void Timer::DoBuild() {
   }
 }
 
+void Timer::PartitionTickers(std::vector<TimeListener*>& cpp_agents, std::vector<TimeListener*>& py_agents) {
+  for (std::pair<int, TimeListener*> pair : tickers_) {
```

Comment:
What if we generate this partition at the same time that we populate `tickers_`? (And at the places when we depopulate it for decommissioning?)

This avoids having to regenerate it at each Tick

Reply:
I added std::vectors `cpp_tickers_` and `py_tickers_` that are registered/unregistered at the same time as `tickers_`. I opted to use a vector so we don't have to transform it from std::map to something that is parallelizable at each timestep